Use docker container as local SSL proxy

finkef opened the pull request:
This PR replaces the `local-ssl-proxy` npm package with a Docker container image for SSL termination during testing. Apparently, there is an issue in the `http` package that causes a proxy server to hang or cause file descriptor limit issues.

More details can be found in this discussion: https://github.com/vitejs/vite/issues/11468#issuecomment-1364355483

raineorshine reviewed (changes requested):
The puppeteer tests seem to start out okay on my local machine, but at a certain point they start failing with ` 502 (Bad Gateway)`. Are you getting the same?

```sh
stdout | src/e2e/puppeteer/__tests__/cursor.ts > do nothing when clicking on a hidden ancestor
Failed to load resource: the server responded with a status of 502 (Bad Gateway)
Failed to load resource: the server responded with a status of 502 (Bad Gateway)
Failed to load resource: the server responded with a status of 502 (Bad Gateway)
Failed to load resource: the server responded with a status of 502 (Bad Gateway)
Failed to load resource: the server responded with a status of 502 (Bad Gateway)

stdout | src/e2e/puppeteer/__tests__/cursor.ts > do nothing when clicking on a hidden ancestor
Failed to load resource: the server responded with a status of 502 (Bad Gateway)

stdout | src/e2e/puppeteer/__tests__/cursor.ts > do nothing when clicking on a hidden ancestor
Failed to load resource: the server responded with a status of 502 (Bad Gateway)
WebSocket connection to 'wss://localhost:3000/' failed: Error in connection establishment: net::ERR_SSL_PROTOCOL_ERROR
[vite] failed to connect to websocket.
your current setup:
  (browser) host.docker.internal:3001/ <--[HTTP]--> localhost:3000/ (server)
  (browser) host.docker.internal:3001/ <--[WebSocket (failing)]--> localhost:3000/ (server)
Check out your Vite / network configuration and https://vitejs.dev/config/server-options.html#server-hmr .

stdout | src/e2e/puppeteer/__tests__/caret.ts > all platforms > on cursorDown, the caret should move to the beginning of the new cursor.
Failed to load resource: the server responded with a status of 502 (Bad Gateway)
```

raineorshine commented:
I reverted 41f33ae5600062e3b2be36da6b9689172e69329a (from #2339) on `main` until we can get a stable build.

But reapplied it on this PR so it effectively was not reverted.

finkef commented:
@raineorshine I could not reproduce the issues mentioned, but we should probably drop the local proxy in favor of just letting vite handle it.

I've adjusted the script to spawn a separate https dev server _without_ HMR which seemed to be problematic and also disabled the `vite-error-overlay` which was annoying during puppeteer runs with unused variables. Also moved the puppeteer dev server to a random, unused port.

So far, couldn't produce any more issues locally.

During Github actions, we run against a static production build, same as the original strategy went. We could easily run against a dev server here too, but I believe testing against production is much more sensible during CI.

raineorshine reviewed (approved):
Works great! This clears up all the 503 errors I was getting on my local machine.
